tutorial/chapter1.md

```diff
@@ -0,0 +1,200 @@
+            std::cout << "ImageProcessingService_5 - result data received: " << message.mean << "; number of data received: " << ctrImageProcessingServices[4] <<std::endl;
+        }
+    );
+   
+    imageProcessingService_1.startup();
+    imageProcessingService_2.startup();
+    imageProcessingService_3.startup();
+    imageProcessingService_4.startup();
+    imageProcessingService_5.startup();
+    fileImageStreamingService.startup();
+
+    ServerUserInput serverUserInput;
+
+    std::thread imageProcessingServiceThread = std::thread([&]() {
+        while (true) {
+            fileImageStreamingService.runOnce();
+            imageProcessingService_1.runOnce();
+            imageProcessingService_2.runOnce();
+            imageProcessingService_3.runOnce();
+            imageProcessingService_4.runOnce();
+            imageProcessingService_5.runOnce();
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            if (!serverUserInput._isRunning) {
+                break;
+            }
+        }
+    });
+
+    serverUserInput.run();
+
+    imageProcessingServiceThread.join();
+
+    fileImageStreamingService.shutdown();
+    imageProcessingService_5.shutdown();
+    imageProcessingService_4.shutdown();
+    imageProcessingService_3.shutdown();
+    imageProcessingService_2.shutdown();
+    imageProcessingService_1.shutdown();
+
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_5")->removeListener("image_processing_results_5");
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_4")->removeListener("image_processing_results_4");
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_3")->removeListener("image_processing_results_3");
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_2")->removeListener("image_processing_results_2");
+    eventloop.getSubscriber<kpsr_tutorial::ResultData>("image_processing_results_1")->removeListener("image_processing_results_1");
+
+    eventloop.stop();
+
+
+    for (int i = 0; i < nImageProcessingServices; i++)
+    {
+        std::cout << "Total number of data received in imageProcessingService_" << i+1 << ": "<< ctrImageProcessingServices[i] <<std::endl;
+    }  
+}
+```
+In this example there are some features of Klepsydra that will be introduced and others that have been described in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
+
+* From the module kpsr-robotics/vision_ocv/modules/mem_mdlw, the ```kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider``` class is introduced. Note that a param of this class is ```kpsr::high_performance::DataMultiplexerMiddlewareProvider```, which is a single-producer, multiple-consumer introduced in the [kpsr-tutorial - chapter 1](https://github.com/klepsydra-technologies/kpsr-tutorial/blob/master/tutorials/chapter1.md).
```

Comment:
The ```kpsr::vision_ocv::ImageMultiplexerVisionMiddlewareProvider``` class introduced here uses the ImageDataFactory and the DataMutltiplexerMiddlewareProvider. The ImageDataFactory handles the allocation... (etc) which are used by the underlying DataMutliplexer... to ensure that the memory allocation, initialization and copies are done correctly.